content/en/blog/_posts/2022-11-XX-seccomp-notifier.md

```diff
@@ -0,0 +1,347 @@
+guarded by an annotation, which has to be explicitly allowed, for example by
+using a configuration drop-in like this:
+
+```shell
+> cat /etc/crio/crio.conf.d/02-runtimes.conf
+```
+
+```toml
+[crio.runtime]
+default_runtime = "runc"
+
+[crio.runtime.runtimes.runc]
+allowed_annotations = [ "io.kubernetes.cri-o.seccompNotifierAction" ]
+```
+
+[sources]: https://github.com/cri-o/cri-o/blob/main/install.md#build-and-install-cri-o-from-source
+[script]: https://github.com/cri-o/cri-o#installing-cri-o
+
+If CRI-O is up and running, then it should indicate that the seccomp notifier is
+available as well:
+
+```shell
+> sudo ./bin/crio --enable-metrics
+…
+INFO[…] Starting seccomp notifier watcher
+INFO[…] Serving metrics on :9090 via HTTP
+…
+```
+
+We also enable the metrics, because they provide additional telemetry data about
+the notifier. Now we need a running Kubernetes cluster for demonstration
+purposes. For this demo, we mainly stick to the
+[`hack/local-up-cluster.sh`][local-up] approach to locally spawn a single node
+Kubernetes cluster.
+
+[local-up]: https://github.com/cri-o/cri-o#running-kubernetes-with-cri-o
+
+If everything is up and running, then we would have to define a seccomp profile
+for testing purposes. But we do not have to create our own, we can just use the
+`RuntimeDefault` profile which gets shipped with each container runtime. For
+example the `RuntimeDefault` profile for CRI-O can be found in the
+[containers/common][runtime-default] library.
+
+[runtime-default]: https://github.com/containers/common/blob/afff1d6/pkg/seccomp/seccomp.json
+
+Now we need a test container, which can be a simple [nginx][nginx] pod like
+this:
+
+[nginx]: https://www.nginx.com
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  annotations:
+    io.kubernetes.cri-o.seccompNotifierAction: "stop"
+spec:
+  restartPolicy: Never
+  containers:
+    - name: nginx
+      image: nginx:1.23.2
+      securityContext:
+        seccompProfile:
+          type: RuntimeDefault
+```
+
+Please note the annotation `io.kubernetes.cri-o.seccompNotifierAction`, which
+enables the seccomp notifier for this workload. The value of the annotation can
+be either `stop` for stopping the workload or anything else for doing nothing
+else than logging and throwing metrics. Because of the termination we also use
+the `restartPolicy: Never` to not automatically recreate the container on
+failure.
+
+Let's run the pod and check if it works:
+
+```shell
+> kubectl apply -f nginx.yaml
+```
+
+```shell
+> kubectl get pods -o wide
+NAME    READY   STATUS    RESTARTS   AGE     IP          NODE        NOMINATED NODE   READINESS GATES
+nginx   1/1     Running   0          3m39s   10.85.0.3   127.0.0.1   <none>           <none>
+```
+
+We can also test if the web server itself works as intended:
+
+```shell
+> curl 10.85.0.3
+<!DOCTYPE html>
+<html>
+<head>
+<title>Welcome to nginx!</title>
+…
```

Comment:
This isn't a valid shell script. I recommend changing the format indicator on the previous line.

Imagine that we want one day to lint these scripts. We probably don't, but we do want to be able to change the highlighter without checking all of the pages that might misuse it.

Reply:
Alright, I changed it to `console`.